src/logic/contracts/safeContracts.ts

```diff
@@ -268,7 +269,7 @@ export const estimateGasForDeployingSafe = async (
 }
 
 export const getGnosisSafeInstanceAt = (safeAddress: string, safeVersion: string): GnosisSafe => {
-  const safeSingletonDeployment = getSafeContractDeployment({ safeVersion })
+  const safeSingletonDeployment = getSafeContractDeployment({ safeVersion: semverClean(safeVersion) })
```

Comment:
Probably not too critical but `semverClean` can return `null`. We should consider installing `@types/semver`

Reply:
Nice catch. `getGnosisSafeInstanceAt` is used in a lot of places so we'd have the adjust far more than just these two files.

I'll leave this as is for the time being as it's a draft but we can address that if we agree that this is the best approach.

Reply:
I think it's weird (and wrong) that `semverClean` removes the "+L2"
According to the semver docs:

>`semverClean`: Clean a string to be a valid semver if possible

But according to the semver spec: https://semver.org/#spec-item-10 appending numerics and characters with a "+" is absolutely valid. So why does semverClean remove this?
I'm just a bit worried, that this might get changed in future semver package updates.

Also there is another call to `getSafeContractDeployment` so we could move this semver  cleaning in that function.

Reply:
Also the specs say:
> Build metadata MUST be ignored when determining version precedence

So I would have expected semverSatisfies("1.3.0+L2" , ">=1.3.0"); to ignore the build metadata. Not sure if this is a bug in that package or if I misunderstand the spec here.

Reply:
> I think it's weird (and wrong) that semverClean removes the "+L2"

It works as expected. It 'cleans' the meta as it has no "version precedence".

> I would have expected semverSatisfies("1.3.0+L2" , ">=1.3.0"); to ignore the build metadata.

The deployments repo strictly compares versions (`===`) and doesn't use `semver`. This is the main reason behind the bug.

> Also there is another call to getSafeContractDeployment so we could move this semver cleaning inside getSafeContractDeployment.

The other `getSafeContractDeployment` call references the `LATEST_SAFE_VERSION` and is therefore not an issue here.

Whilst I agree that this is likely better done in the deployments repo, let's discuss it in the retro tomorrow. I have added my thoughts to the postmortem doc already. To summarise them though:

- The deployments repo currently has no dependencies. Do we want to add one when it is fixed here?
- The getter have an optional filter which [doesn't have a version fallback](https://github.com/safe-global/safe-deployments/blob/32268ae8987380498a458da46b7691e8cd599f2a/src/utils.ts#L12).
- Would we then want to keep two separate getters or rely purely on the meta?